ui/mirage/factories/job.js

```diff
@@ -200,6 +200,11 @@ export default Factory.extend({
   shallow: false,
 
   afterCreate(job, server) {
+    Ember.assert(
+      '[Mirage] No node pools! make sure node pools are created before jobs',
+      server.db.nodePools.length
+    );
```

Comment:
Is there always a `default` node pool, in practice, to keep this from being length=0 in real life?

Reply:
Yup! There are even 2 node pools created automatically:
https://github.com/hashicorp/nomad/blob/04042c619d68ad1f67da87c3c0c449de0f9d0637/nomad/state/state_store_node_pools.go#L13-L31

And just to clarify, this assertion is only applied on dev environments using Mirage, so there's no end user impact.